src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -421,6 +425,26 @@ InterpreterSelectQuery::InterpreterSelectQuery(
         RewriteCountDistinctFunctionVisitor(data_rewrite_countdistinct).visit(query_ptr);
     }
 
+    /// proton: porting starts. TODO: remove this comment. Try to process the streaming query extension grammar.
+    /// we need to process before table storage generation (maybe has table function)
+    if (auto emit = getSelectQuery().emit())
```

Comment:
Thanks @KochetovNicolai for the comment. We will take a look.

Reply:
@KochetovNicolai may we resolve this analyzer issue in an immediate following PR since this one is big enough and we will need some time to sort out  how to work with the new analyzer ? Thanks !